### PR TITLE
curvefs: support set metaserver offline

### PR DESCRIPTION
**Pull request by SeanHai**
Signed-off-by: wanghai01 <seanhaizi@163.com>

### What problem does this PR solve?

Issue Number: #xxx 
#1838 
Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/test/mds/topology/test_topology.cpp [hunk 1]
```diff
@@ -2340,6 +2340,29 @@ TEST_F(TestTopology, GenCopysetAddrBatch_2pool) {
...
+        0x32, "server2", "127.0.0.2" , 0, "127.0.0.2" , 0, 0x22, 0x11);
+    PrepareAddServer(
+        0x33, "server3", "127.0.0.3" , 0, "127.0.0.3" , 0, 0x23, 0x11);
+    PrepareAddMetaServer(0x41, "metaserver1", "token1", 0x31,
+                        "127.0.0.1", 8200, "127.0.0.1", 8200);
+    PrepareAddMetaServer(0x42, "metaserver2", "token2", 0x31,
+                        "127.0.0.1", 8200, "127.0.0.1", 8200);
+
+    CopySetInfo copysetInfo;
+    ASSERT_TRUE(topology_->IsMetaServerReRegistered(0x42));
```
Comment (cw123) [inline curvefs/test/mds/topology/test_topology.cpp:2363]: add a false case

File: curvefs/test/mds/topology/test_topology_manager.cpp [hunk 2]
```diff
@@ -2842,6 +2842,26 @@ TEST_F(TestTopologyManager, test_GetCopysetMembers_Success) {
...
+        "ip2", 8887);
+
+    OfflineMetaServerRequest request;
+    request.set_metaserverid(msId);
+
+    OfflineMetaServerResponse response;
+
+    serviceManager_->OfflineMetaServer(&request, &response);
+    ASSERT_EQ(TopoStatusCode::TOPO_OK, response.statuscode());
+}
```
Comment (cw123) [inline curvefs/test/mds/topology/test_topology_manager.cpp:2863]: check if the metaserver is offline

**Comment by cw123** [comment]
Paste test screenshot. prove this pr can solve the https://github.com/opencurve/curve/issues/1838

File: curvefs/src/mds/topology/topology.cpp [hunk 3]
```diff
@@ -536,6 +536,23 @@ bool TopologyImpl::GetMetaServer(const std::string &hostIp, uint32_t port,
...
+bool TopologyImpl::IsMetaServerReRegistered(MetaServerIdType msId) {
+    MetaServer ms;
+    if (GetMetaServer(msId, &ms)) {
+        ReadLockGuard rlockMetaServerMap(metaServerMutex_);
+        for (auto it = metaServerMap_.begin(); it != metaServerMap_.end();
+            it++) {
+            ReadLockGuard rlockMetaServer(it->second.GetRWLockRef());
+            if (it->second.GetInternalIp() == ms.GetInternalIp() &&
+                it->second.GetInternalPort() == ms.GetInternalPort() &&
+                it->second.GetId() != ms.GetId()) {
```
Comment (Cyber-SiKu) [inline curvefs/src/mds/topology/topology.cpp:548]: Why not judge external ip/prot.

**Review by Cyber-SiKu — CHANGES_REQUESTED** [review]
Please implement it in tools-v2 instead of old tools.

**Pull request closed without merging**